### ISSUE TITLE
Apache Fullstatus broken via Curl

### ISSUE THREAD

**Issue by LukeHandle** [rackerlabs/recap#125]
Hi there,

It appears that changes were made to remove `elinks` requirements from spec/rpm:
https://src.fedoraproject.org/rpms/recap/blob/master/f/recap.spec (as not everyone needs Apache fullstatus, I understand the thought process)

And thus, the project switched to use a `curl` instead of `apachectl fullstatus` (`e/links`). This is not adequate for reporting Apache server-status though:

```
Web Status report
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<html><head>
<title>Apache Status</title>
</head><body>
<h1>Apache Server Status for localhost</h1>

<dl><dt>Server Version: Apache/2.2.15 (Unix) DAV/2 PHP/5.4.43 mod_ssl/2.2.15 OpenSSL/1.0.1e-fips mod_perl/2.0.4 Perl/v5.10.1</dt>
<dt>Server Built: Jul 23 2014 07:06:18
</dt></dl><hr /><dl>
<dt>Current Time: Thursday, 12-Oct-2017 16:23:12 CEST</dt>
<dt>Restart Time: Thursday, 12-Oct-2017 14:42:27 CEST</dt>
<dt>Parent Server Generation: 0</dt>
<dt>Server uptime:  1 hour 40 minutes 45 seconds</dt>
<dt>Total accesses: 171637 - Total Traffic: 1.1 GB</dt>
<dt>CPU Usage: u396.22 s220.44 cu0 cs0 - 10.2% CPU load</dt>
<dt>28.4 requests/sec - 194.9 kB/second - 6.9 kB/request</dt>
<dt>72 requests currently being processed, 14 idle workers</dt>
</dl><pre>KKWCCKCCKCKWCKKKK_KC.K....CWKKKKK.._C._._.._.C.K..__C_..W.KK....
K....KWKK...KK..CKWK_CKCW...C.K.CK....K_CKCK.K.CKC.C.__KKW..W..K
.__WWWW.........................................................
................................................................
................................................................
................................................................
................................................................
................................................................
</pre>
<p>Scoreboard Key:<br />
"<b><code>_</code></b>" Waiting for Connection,
"<b><code>S</code></b>" Starting up,
"<b><code>R</code></b>" Reading Request,<br />
"<b><code>W</code></b>" Sending Reply,
"<b><code>K</code></b>" Keepalive (read),
"<b><code>D</code></b>" DNS Lookup,<br />
"<b><code>C</code></b>" Closing connection,
"<b><code>L</code></b>" Logging,
"<b><code>G</code></b>" Gracefully finishing,<br />
"<b><code>I</code></b>" Idle cleanup of worker,
"<b><code>.</code></b>" Open slot with no current process</p>
<p />


<table border="0"><tr><th>Srv</th><th>PID</th><th>Acc</th><th>M</th><th>CPU
</th><th>SS</th><th>Req</th><th>Conn</th><th>Child</th><th>Slot</th><th>Client</th><th>VHost</th><th>Request</th></tr>
[...]
```

I've highlighted this against the commit as well:
https://github.com/rackerlabs/recap/commit/3f4b5204149479eb303afbea81d0f0e972e65fdb#commitcomment-24934576

Further to this, the default status URL here is a bit strange:

https://github.com/rackerlabs/recap/blob/7cbbfb592000c63bdb46824954a7a102a53a2916/src/recap.conf#L216

Traditionally this has been an Apache specific full status option - is does not seem logical to change this default to point to the homepage of default VHost. This should instead default to `http://localhost:80/server-status`.

===

I suggest maybe adding a check for when `e/links` is present to instead run:
`/usr/bin/links -dump http://localhost:80/server-status`

**Comment by tonyskapunk** [comment]
Hi @LukeHandle thanks for opening the issue and providing the details.

You're right the formatting is not the best, I'm looking into it. Sounds like the easiest solution would be implementing`links --dump`, I'm going to do some testing and provide different outputs to see what's best.

The default value for [OPTS_STATUSURL]( https://github.com/rackerlabs/recap/blob/master/src/recap#L101) is generic, in the past recap assumed one would *only* use apache httpd([through apachectl](https://github.com/rackerlabs/recap/commit/3f4b5204149479eb303afbea81d0f0e972e65fdb#diff-cf77cb5672a68c31afd4bc3025d4288cL240)) as the web server where the stats were being pulled from.  Additionally [USEFULLSTATUS](https://github.com/rackerlabs/recap/blob/master/src/recap#L7) by default is set to "no" someone making use of it would have to change it in the config file and now it will have to be changed the `OPT_STATUSURL` to the desired one as well.

**Comment by tonyskapunk** [comment]
It took me a while but I created a gist with the outputs using `apache httpd`, `nginx`, and `lighttpd` with the current `curl` output(unformatted) and using `links -dump`(formatted) which looks a lot better.

https://gist.github.com/tonyskapunk/7fa78554a617fcda6f7ea2810203f939

If no objections are made then the next step is to write the code to switch to use `links`, that change is not a big deal, the caveat is to replace the `OPTS_CURL`[0][1][2] for somethinig like `OPTS_LINKS` or similar.

[0] https://github.com/rackerlabs/recap/blob/master/src/recap#L90
[1] https://github.com/rackerlabs/recap/blob/master/src/recap.conf#L160-L163
[2] https://github.com/rackerlabs/recap/blob/master/src/recap.5#L154-L157

---
cc: @LukeHandle, @man-chung

**Comment by LukeHandle** [comment]
Hey @tonyskapunk, thanks for your time on this, much appreciated!

Looks better, shame about `nginx`, but it's still readable 🤷‍♂️ 

With regards to the default URL, having this set to `http://localhost/server-status` is at least a saner default. No-one would set the stats to be at `/`. This would also  stop people enabling the "USEFULLSTATUS" option and expecting the previous behaviour.

At the very least, updating the comment to specify the default URL (hinting that likely it needs changing further down the config). Currently it's more of a "the URL can be changed", less of a "the URL needs to be changed".

**Comment by man-chung** [comment]
+1 to the comment hint drop, at least. But its better to put the full `http://localhost/server-status` since that would be what any familiar with full status would expect.

**Comment by tonyskapunk** [comment]
After the test performed I noticed lighttpd uses `server-status` as well, and while `nginx` uses `nginx_status` in the examples it can be anything, so I concur, the option default `http://localhost:80/server-status` is better

**Comment by tonyskapunk** [comment]
@LukeHandle , @man-chung would you mind looking t the PR #127 I implemented the changes in there, let me know how it looks like and if there is any feedback related to it.

Thanks!

**Comment by LukeHandle** [comment]
Hey there @tonyskapunk, the README will also need to be updated to include reference to the new dependency of `elinks` (included in base).

Of course, the spec file will need adjusting etc. before build.

**Comment by tonyskapunk** [comment]
@LukeHandle thanks for checking, I added the dependency of `links` in the README for those who install not from a package.  I'll make sure I communicate this to @carlwgeorge about the new dependency around `links` due to the formatting.

**Comment by tonyskapunk** [comment]
Fixed in #127
